### ISSUE TITLE
confusion regarding the reward

### ISSUE THREAD

**Issue by super-shayan** [Thartvigsen/EARLIEST#1]
hello! i'm a bit confused as to how the reward is computed in the code, specifically in the computeLoss function on line 125 in model.py: 
        _, y_hat = torch.max(torch.softmax(logits, dim=1), dim=1)
        self.r = (2*(y_hat.float().round() == y.float()).float()-1).detach().unsqueeze(1)
        self.R = self.r * self.grad_mask
Isnt it that self.r would be constant value reflecting the last predictions the discriminator network has made? thenit gets multiplied with a vector grad_mask, so as to repeat the reward as many times till the "halting point". But why doesn't the reward reflect the intermediate predictions made via the discriminator too?

**Comment by Thartvigsen** [comment]
Hi @super-shayan , this is a good question, you are right that self.r is a constant value. I think your suggestion that the reward should reflect the intermediate predictions made by the discriminator is a good one (and since then, that's what we've done), but in this paper we just used the predictions made by the discriminator at the very last step!
